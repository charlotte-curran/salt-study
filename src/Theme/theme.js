export const SaltTheme = {
  color: {
    dark: "#222324",
    blue: "#92b3d2",
    white: "rgba(255,255,255, 1)",
    orange: "#ff7a61",
    shadow: "#545353",
    address: "rgba(255, 255, 255, 0.4)",
    navbar: "rgba(34, 35, 36, 0.7)"
  },
  fonts: {
    futuraPt: '"futura-pt", sans-serif',
    raleway: '"raleway", sans-serif'
  }
};
