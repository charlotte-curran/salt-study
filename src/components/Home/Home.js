import React, { useRef, useState } from "react";
import SquareOne from "../SquareOne";
import SquareTwo from "../SquareTwo";
import SquareThree from "../SquareThree";
import SquareFour from "../SquareFour";
import SquareFive from "../SquareFive";
import SquareSix from "../SquareSix";
import SquareSeven from "../SquareSeven";
import SquareEight from "../SquareEight";
import SquareNine from "../SquareNine";
import SquareTen from "../SquareTen";
import NavBar from "../NavBar";
import Splashpage from "../SplashPage";
import { Wrapper, WrapperReverse } from "./Home.styles";
import { useLocaleContext } from "../../Context/localeContext";

const Home = () => {
  const secondWrapperRef = useRef(null);
  const squareSixRef = useRef(null);
  const thirdWrapperRef = useRef(null);
  const refObj = {
    secondWrapperRef: secondWrapperRef,
    squareSixRef: squareSixRef,
    thirdWrapperRef: thirdWrapperRef
  };
  const [isVisibleModal, setIsVisibleModal] = useState(false);
  const { locale, content } = useLocaleContext();

  return (
    <div>
      {locale !== 0 ? (
        <div>
          <NavBar
            content={content[11]}
            refObj={refObj}
            isVisibleModal={isVisibleModal}
          />
          <Wrapper>
            <SquareOne content={content[1]} isVisibleModal={isVisibleModal} />
            <SquareTwo
              content={content[2]}
              careerPack={content[12]}
              isVisibleModal={isVisibleModal}
              setIsVisibleModal={setIsVisibleModal}
            />
          </Wrapper>
          <WrapperReverse ref={secondWrapperRef}>
            <SquareThree content={content[3]} />
            <SquareFour content={content[4]} />
          </WrapperReverse>
          <Wrapper>
            <SquareFive content={content[5]} />
          </Wrapper>
          <Wrapper ref={squareSixRef}>
            <SquareSix content={content[6]} />
          </Wrapper>
          <Wrapper ref={thirdWrapperRef}>
            <SquareSeven content={content[7]} />
            <SquareEight content={content[8]} />
          </Wrapper>
          <SquareNine content={content[9]} />
          <SquareTen
            content={content[10]}
            careerPack={content[12]}
            isVisibleModal={isVisibleModal}
            setIsVisibleModal={setIsVisibleModal}
          />
        </div>
      ) : (
        <Splashpage />
      )}
    </div>
  );
};

export default Home;
