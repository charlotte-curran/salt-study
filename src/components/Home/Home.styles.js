import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  /* z-index: 5; */

  @media (min-width: 1000px) {
    flex-direction: row;
  }

  /* ipad pro portrait*/
  @media (width: 1024px) and (height: 1366px) {
    flex-direction: column;
  }
`;

export const WrapperReverse = styled.div`
  @media (min-width: 1000px) {
    display: flex;
    flex-direction: row-reverse;
  }

  /* ipad pro portrait*/
  @media (width: 1024px) and (height: 1366px) {
    display: flex;
    flex-direction: column;
  }
`;
