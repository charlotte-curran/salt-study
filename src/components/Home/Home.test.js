import React from "react";
import { create } from "react-test-renderer";
import Home from "./Home";
import { ThemeProvider } from "styled-components";
import { SaltTheme } from "../../Theme/theme";
import LocaleProvider from "../../Context/localeContext";

describe("Home", () => {
  test("matches the snapshot", () => {
    const home = create(
      <LocaleProvider>
        <ThemeProvider theme={SaltTheme}>
          <Home />
        </ThemeProvider>
      </LocaleProvider>
    );
    expect(home.toJSON()).toMatchSnapshot();
  });
});
