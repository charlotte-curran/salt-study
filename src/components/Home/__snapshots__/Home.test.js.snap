// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div>
  <div
    className="sc-bwCtUz bNnsMZ"
  >
    <img
      className="sc-bxivhb hBHhOs"
      src="saltplus.svg"
    />
    <img
      className="sc-EHOje cPCjJa"
      src="saltplus.svg"
    />
    <img
      className="sc-ifAKCX kVGeYQ"
      src="saltplus.svg"
    />
    <img
      className="sc-bZQynM dUWWhk"
      src="saltplus.svg"
    />
    <div
      className="sc-hrWEMg hDXHrU"
    >
      <div
        className="sc-eTuwsz ipodbl"
      >
        Welcome to 
        <span
          className="sc-gwVKww hVgRf"
        >
          salt
        </span>
        !
      </div>
      <div
        className="sc-hXRMBi JpweF"
      >
        Pick your location:
      </div>
      <div
        className="sc-epnACN bTlTCl"
      >
        <button
          className="sc-htoDjs kGyMxE"
          onClick={[Function]}
        >
          Stockholm
        </button>
        <button
          className="sc-htoDjs kGyMxE"
          onClick={[Function]}
        >
          Amsterdam
        </button>
        <button
          className="sc-htoDjs kGyMxE"
          onClick={[Function]}
        >
          Oslo
        </button>
      </div>
    </div>
  </div>
</div>
`;
